zerver/views/users.py

```diff
@@ -297,12 +302,14 @@ def add_bot_backend(request, user_profile, full_name_raw=REQ("full_name"), short
         user_file = list(request.FILES.values())[0]
         upload_avatar_image(user_file, user_profile, bot_profile)
 
-    if bot_type == UserProfile.OUTGOING_WEBHOOK_BOT:
-        add_outgoing_webhook_service(name=service_name,
-                                     user_profile=bot_profile,
-                                     base_url=payload_url,
-                                     interface=1,
-                                     token=random_api_key())
+    if bot_type == UserProfile.OUTGOING_WEBHOOK_BOT or bot_type == UserProfile.EMBEDDED_BOT:
+        if bot_type == UserProfile.EMBEDDED_BOT:
+            payload_url = ""
+        add_service(name=service_name,
+                    user_profile=bot_profile,
+                    base_url=payload_url,
+                    interface=1,
+                    token=random_api_key())
```

Comment:
For Embedded bots, we should probably check whether the bot is in our registry, and if not, return an error.